pandas/core/generic.py

```diff
@@ -6815,16 +6815,21 @@ def mask(self, cond, other=np.nan, inplace=False, axis=None, level=None,
                           errors=errors)
 
     _shared_docs['shift'] = ("""
-        Shift index by desired number of periods with an optional time freq
+        Shift index by desired number of periods with an optional time freq.
+
+        When freq is not passed, shift the index without realign the data.
+        If freq is passed (in this case, the index must be date or datetime),
+        the index will be increased using the periods and the freq.
 
         Parameters
         ----------
         periods : int
-            Number of periods to move, can be positive or negative
+            Number of periods to move, can be positive or negative.
```

Comment:
Very minor but you want a semicolon instead of a comma here. Perhaps saying "shift" instead of move would be better